bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/observability/ObserverContext.java

```diff
@@ -73,9 +73,9 @@
     private boolean isSystemSpan;
 
     public ObserverContext() {
-        this.properties = new HashMap<>();
-        this.mainTags = new HashMap<>();
-        this.additionalTags = new HashMap<>();
+        this.properties = new ConcurrentHashMap<>();
```

Comment:
do we need to change this to ConcurrentHashMap ?

Reply:
Since this had not been an issue so far, I have reverted this change with 62949558fcaffac152102ab246d94114db89cf97. I will verify this further and add back the ConcurrentHashMaps later if needed.